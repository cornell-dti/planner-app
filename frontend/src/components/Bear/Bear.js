import React, { Component } from 'react'
import { connect } from 'react-redux'
import { updateBearStatus } from '../../store/actions.js'

const mapStateToProps = state => {
    return { bearStatus: state.bearStatus };
};

class Bear extends Component {
    render() {
        return (
            <div>
                {this.state.bearStatus}
            </div>
        );
    }
};

const GetBearStatus = connect(mapStateToProps, null)(Bear);
export default GetBearStatus
