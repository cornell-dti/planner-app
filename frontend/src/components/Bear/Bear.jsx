import React, { Component } from 'react';
import { connect } from 'react-redux';

const mapStateToProps = state => ({ bearStatus: state.bearStatus });

class Bear extends Component {
  render() {
    const { bearStatus } = this.state;
    return <div>{ bearStatus }</div>;
  }
}

const GetBearStatus = connect(mapStateToProps, null)(Bear);
export default GetBearStatus;
