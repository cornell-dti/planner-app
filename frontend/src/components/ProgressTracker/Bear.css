.Bear {
  width: 50px;
  background-size: 50px auto;
  margin-left: 1em;
}
