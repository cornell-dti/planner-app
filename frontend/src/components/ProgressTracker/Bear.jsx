// @flow strict

import React from 'react';
import type { Node } from 'react';
import type { TasksProgressProps } from '../../util/task-util';
import HappyBear from '../../assets/bear/happy-bear.png';
import RegularBear from '../../assets/bear/regular-bear.png';
import styles from './Bear.module.css';

/**
 * The bear as a progress checker.
 */
export default function Bear({ completedTasksCount, allTasksCount }: TasksProgressProps): Node {
  const allCompleted = completedTasksCount === allTasksCount;
  const style = { backgroundImage: `url(${allCompleted ? HappyBear : RegularBear})` };
  return (
    <div className={styles.Bear} style={style} />
  );
}
