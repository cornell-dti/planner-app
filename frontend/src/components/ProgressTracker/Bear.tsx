import React, { ReactElement } from 'react';
import { TasksProgressProps } from '../../util/task-util';
import HappyBear from '../../assets/bear/happy-bear.png';
import RegularBear from '../../assets/bear/regular-bear.png';
import styles from './Bear.css';

/**
 * The bear as a progress checker.
 */
export default ({ completedTasksCount, allTasksCount }: TasksProgressProps): ReactElement => {
  const allCompleted = completedTasksCount === allTasksCount;
  const style = { backgroundImage: `url(${allCompleted ? HappyBear : RegularBear})` };
  return <div className={styles.Bear} style={style} />;
};
