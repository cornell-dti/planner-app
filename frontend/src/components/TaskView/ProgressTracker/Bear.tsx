import React, { ReactElement, CSSProperties } from 'react';
import { TasksProgressProps } from 'common/util/task-util';
import styles from './Bear.module.scss';
import { HappyBear, RegularBear } from '../../../assets/assets-constants';

type Props = TasksProgressProps & { readonly inMobileView: boolean };

const Bear = ({ completedTasksCount, allTasksCount, inMobileView }: Props): ReactElement => {
  const allCompleted = completedTasksCount === allTasksCount;
  let style: CSSProperties = { backgroundImage: `url(${allCompleted ? HappyBear : RegularBear})` };
  if (!inMobileView) {
    style = { ...style, margin: '0' };
  }
  return <div className={styles.Bear} style={style} />;
};

/**
 * The bear as a progress checker.
 */
export default Bear;
