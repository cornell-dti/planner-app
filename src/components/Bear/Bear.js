import React, { Component } from 'react'
import { connect } from 'reac-redux'
import { updateBearStatus } from '../store/actions.js'

const mapStateToProps = state => {
    return { bearStatus: state.bearStatus };
};

const mapDispatchToProps = dispatch => {
    return {
        updateBearStatus: () => dispatch(updateBearStatus());
    };
};

class Bear extends Component {
    render() {
        return (
            <div>
                {this.state.bearStatus}
            </div>
        );
    }
};

const GetBearStatus = connect(mapStateToProps, null)(Bear);
export default GetBearStatus