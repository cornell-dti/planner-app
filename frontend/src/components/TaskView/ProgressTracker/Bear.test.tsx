import React from 'react';
import renderer from 'react-test-renderer';
import Bear from './Bear';

it('Mobile completed bear matches snapshot', () => {
  const tree = renderer.create(
    <Bear inMobileView allTasksCount={3} completedTasksCount={3} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Mobile incompleted bear matches snapshot', () => {
  const tree = renderer.create(
    <Bear inMobileView allTasksCount={1} completedTasksCount={3} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Desktop completed bear matches snapshot', () => {
  const tree = renderer.create(
    <Bear inMobileView={false} allTasksCount={3} completedTasksCount={3} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Desktop incompleted bear matches snapshot', () => {
  const tree = renderer.create(
    <Bear inMobileView={false} allTasksCount={1} completedTasksCount={3} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
