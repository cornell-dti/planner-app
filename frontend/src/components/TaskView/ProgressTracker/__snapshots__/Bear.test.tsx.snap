// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Desktop completed bear matches snapshot 1`] = `
<div
  className="Bear"
  style={
    Object {
      "backgroundImage": "url(/bear/happy-bear.png)",
      "margin": "0",
    }
  }
/>
`;

exports[`Desktop incompleted bear matches snapshot 1`] = `
<div
  className="Bear"
  style={
    Object {
      "backgroundImage": "url(/bear/regular-bear.png)",
      "margin": "0",
    }
  }
/>
`;

exports[`Mobile completed bear matches snapshot 1`] = `
<div
  className="Bear"
  style={
    Object {
      "backgroundImage": "url(/bear/happy-bear.png)",
    }
  }
/>
`;

exports[`Mobile incompleted bear matches snapshot 1`] = `
<div
  className="Bear"
  style={
    Object {
      "backgroundImage": "url(/bear/regular-bear.png)",
    }
  }
/>
`;
